### PR TITLE
Add ETH-B

### PR DESCRIPTION
**Pull request by brianmcmichael**
Adds ETH-B collateral type to Kovan deployment

### CODE REVIEW

File: src/Kovan-DssSpell.sol [hunk 1]
```diff
@@ -49,23 +49,15 @@ contract SpellAction {
...
     address constant FAUCET          = 0x57aAeAE905376a4B1899bA81364b4cE2519CBfB3;
 
-    // COMP-A specific addresses
-    address constant COMP            = 0x1dDe24ACE93F9F638Bfd6fCE1B38b842703Ea1Aa;
-    address constant MCD_JOIN_COMP_A = 0x16D567c1F6824ffFC460A11d48F61E010ae43766;
-    address constant MCD_FLIP_COMP_A = 0x2917a962BC45ED48497de85821bddD065794DF6C;
-    address constant PIP_COMP        = 0xcc10b1C53f4BFFEE19d0Ad00C40D7E36a454D5c4;
+    // ETH-B specific addresses
+    //   > seth --to-bytes32 $(seth --from-ascii "ETH-B")
+    //   0x4554482d42000000000000000000000000000000000000000000000000000000
```
Comment (lucas-manuel) [inline src/Kovan-DssSpell.sol:54]: Do we need this comment?

File: src/Kovan-DssSpell.sol [hunk 2]
```diff
@@ -80,88 +72,87 @@ contract SpellAction {
...
+        // !!!!!!!! Only if PIP_TOKEN = Osm or PIP_TOKEN = Median and hasn't been already whitelisted due a previous deployed ilk
+        //OsmAbstract(PIP_TOKEN).kiss(MCD_END);
+        // Set TOKEN Osm in the OsmMom for new ilk
+        // !!!!!!!! Only if PIP_TOKEN = Osm
+        OsmMomAbstract(OSM_MOM).setOsm(ilk, PIP_ETH);
 
         // Set the global debt ceiling
-        VatAbstract(MCD_VAT).file("Line", 1196 * MILLION * RAD);
-
+        VatAbstract(MCD_VAT).file("Line", 1216 * MILLION * RAD);  // FIXME
```
Comment (lucas-manuel) [inline src/Kovan-DssSpell.sol:131]: Does this still need to be fixed?

File: src/Kovan-DssSpell.sol (hunk 1 above)
Comment (lucas-manuel) [reply src/Kovan-DssSpell.sol:54]: 👍

**Review by lucas-manuel — APPROVED** [review]
Just checked: 
- Addresses
- Rely/deny on join and flip
- Rate calc
- Tests passing

LGTM

File: src/Kovan-DssSpell.sol [hunk 3]
```diff
@@ -80,88 +72,87 @@ contract SpellAction {
...
+        // Set the TOKEN-LETTER max auction duration to
+        FlipAbstract(MCD_FLIP_ETH_B).file("tau", 1 hours); // 6 hours mainnet
+        // Set the TOKEN-LETTER min collateralization ratio (e.g. 150% => X = 150)
+        SpotAbstract(MCD_SPOT).file(ilk, "mat", 130 * RAY / 100);
+
+        // Update TOKEN-LETTER spot value in Vat
+        SpotAbstract(MCD_SPOT).poke(ilk);
+
+				// Add new ilk to the IlkRegistry
+				IlkRegistryAbstract(ILK_REGISTRY).add(MCD_JOIN_ETH_B);
```
Comment (kmbarry1) [inline src/Kovan-DssSpell.sol:155]: formatting

File: src/Kovan-DssSpell.sol [hunk 4]
```diff
@@ -80,88 +72,87 @@ contract SpellAction {
...
```
Comment (kmbarry1) [inline src/Kovan-DssSpell.sol:156]:
```suggestion
```

File: src/Kovan-DssSpell.sol [hunk 5]
```diff
@@ -80,89 +72,87 @@ contract SpellAction {
...
+        CatAbstract(MCD_CAT).file(ilk, "flip", MCD_FLIP_ETH_B);
+
+        // Init TOKEN-LETTER ilk in Vat & Jug
+        VatAbstract(MCD_VAT).init(ilk);
+        JugAbstract(MCD_JUG).init(ilk);
+
+        // Allow TOKEN-LETTER Join to modify Vat registry
+        VatAbstract(MCD_VAT).rely(MCD_JOIN_ETH_B);
+				// Allow the TOKEN-LETTER Flipper to reduce the Cat litterbox on deal()
+				CatAbstract(MCD_CAT).rely(MCD_FLIP_ETH_B);
```
Comment (kmbarry1) [inline src/Kovan-DssSpell.sol:103]: more tabs

File: src/Kovan-DssSpell.sol [hunk 6]
```diff
@@ -80,89 +72,87 @@ contract SpellAction {
...
+        // Allow Cat to kick auctions in TOKEN-LETTER Flipper
+        FlipAbstract(MCD_FLIP_ETH_B).rely(MCD_CAT);
+        // Allow End to yank auctions in TOKEN-LETTER Flipper
+        FlipAbstract(MCD_FLIP_ETH_B).rely(MCD_END);
+        // Allow FlipperMom to access to the TOKEN-LETTER Flipper
+        FlipAbstract(MCD_FLIP_ETH_B).rely(FLIPPER_MOM);
+				// Disallow Cat to kick auctions in TOKEN-LETTER Flipper
+				// !!!!!!!! Only for certain collaterals that do not trigger liquidations like USDC-A)
```
Comment (kmbarry1) [inline src/Kovan-DssSpell.sol:111]: more tabs

File: src/Kovan-DssSpell.sol [hunk 7]
```diff
@@ -80,89 +72,87 @@ contract SpellAction {
...
+        // Allow OsmMom to access to the TOKEN Osm
+        // !!!!!!!! Only if PIP_TOKEN = Osm and hasn't been already relied due a previous deployed ilk
+        //OsmAbstract(PIP_TOKEN).rely(OSM_MOM);
+        // Whitelist Osm to read the Median data (only necessary if it is the first time the token is being added to an ilk)
+        // !!!!!!!! Only if PIP_TOKEN = Osm, its src is a Median and hasn't been already whitelisted due a previous deployed ilk
+        //MedianAbstract(OsmAbstract(PIP_TOKEN).src()).kiss(PIP_TOKEN);
+        // Whitelist Spotter to read the Osm data (only necessary if it is the first time the token is being added to an ilk)
+        // !!!!!!!! Only if PIP_TOKEN = Osm or PIP_TOKEN = Median and hasn't been already whitelisted due a previous deployed ilk
+        //OsmAbstract(PIP_TOKEN).kiss(MCD_SPOT);
+				// Whitelist End to read the Osm data (only necessary if it is the first time the token is being added to an ilk)
```
Comment (kmbarry1) [inline src/Kovan-DssSpell.sol:123]: more tabs

**Review by kmbarry1 — APPROVED** [review]
LGTM

Only did some spot checks, so don't trust this review too much :).

File: src/Kovan-DssSpell.sol [hunk 8]
```diff
@@ -80,89 +72,87 @@ contract SpellAction {
...
+        /*** ETH-B Collateral Onboarding ***/
+        bytes32 ilk = "ETH-B";
+
+        // Sanity checks
+        require(GemJoinAbstract(MCD_JOIN_ETH_B).vat() == MCD_VAT, "join-vat-not-match");
+        require(GemJoinAbstract(MCD_JOIN_ETH_B).ilk() == ilk, "join-ilk-not-match");
+        require(GemJoinAbstract(MCD_JOIN_ETH_B).gem() == ETH, "join-gem-not-match");
+        require(GemJoinAbstract(MCD_JOIN_ETH_B).dec() == 18, "join-dec-not-match");
+        require(FlipAbstract(MCD_FLIP_ETH_B).vat() == MCD_VAT, "flip-vat-not-match");
+				require(FlipAbstract(MCD_FLIP_ETH_B).cat() == MCD_CAT, "flip-cat-not-match");
```
[the remark this answers is not in the document]
Comment (brianmcmichael) [reply src/Kovan-DssSpell.sol]: This is an artifact of copying the onboarding script from notion. We've added a card to fix this.

**Review by lucas-manuel — APPROVED** [review]

**Pull request merged**